src/Kiota.Builder/Writers/TypeScript/TypeScriptConventionService.cs

```diff
@@ -182,4 +277,69 @@ private string GetDeprecationComment(IDeprecableElement element)
         var removalComment = element.Deprecation.RemovalDate is null ? string.Empty : $" and will be removed {element.Deprecation.RemovalDate.Value.Date.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture)}";
         return $"@deprecated {element.Deprecation.GetDescription(type => GetTypeString(type, (element as CodeElement)!))}{versionComment}{dateComment}{removalComment}";
     }
+
+    public static string GetFactoryMethodName(CodeTypeBase targetClassType, CodeElement currentElement, LanguageWriter? writer = null)
+    {
+        var composedType = GetOriginalComposedType(targetClassType);
+        string targetClassName = composedType is not null ? TranslateType(composedType) : TranslateTypescriptType(targetClassType);
+        var resultName = $"create{targetClassName.ToFirstCharacterUpperCase()}FromDiscriminatorValue";
+        if (GetTypescriptTypeString(targetClassType, currentElement, false, writer) is string returnType && targetClassName.EqualsIgnoreCase(returnType)) return resultName;
+        if (targetClassType is CodeType currentType && currentType.TypeDefinition is CodeInterface definitionClass && GetFactoryMethod(definitionClass, resultName) is { } factoryMethod)
+        {
+            var methodName = GetTypescriptTypeString(new CodeType { Name = resultName, TypeDefinition = factoryMethod }, currentElement, false);
+            return methodName.ToFirstCharacterUpperCase();// static function is aliased
+        }
+        throw new InvalidOperationException($"Unable to find factory method for {targetClassType}");
+    }
+
+    public static T? GetParentOfTypeOrNull<T>(CodeElement codeElement) where T : class
+    {
+        ArgumentNullException.ThrowIfNull(codeElement);
+        try
+        {
+            return codeElement.GetImmediateParentOfType<T>();
```

Comment:
I am using it for a specific scenario in the `GetFactoryMethod` function, the function searches within the CodeFile. If the method is not found there, the search extends to the namespace.
``` c#
private static CodeFunction? GetFactoryMethod(CodeInterface definitionClass, string factoryMethodName)
{
        return GetParentOfTypeOrNull<CodeFile>(definitionClass)?.FindChildByName<CodeFunction>(factoryMethodName)
            ?? GetParentOfTypeOrNull<CodeNamespace>(definitionClass)?.FindChildByName<CodeFunction>(factoryMethodName);
}
```

Reply:
Catching and throwing exceptions can be expensive...
I believe the possible parent of a `codeInterface` should already be known in advance(a CodeFile) since this is setup by the refiner by adding it as a child. 
Would it make sense to then simply do a search from the parent namespace? Is there a reason this would not exist in the codeFile?